src/Components/Components/src/Routing/RouteTable.cs

```diff
@@ -1,132 +1,22 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Reflection;
-using Microsoft.AspNetCore.Components;
-
 namespace Microsoft.AspNetCore.Components.Routing
 {
-    internal class RouteTable
+    internal readonly struct RouteTable
```

Comment:
Is there any benefit to making this a `struct`? Do we ever have more than one instance of it?

(Maybe I'll find out the answer later in this code review...)

Reply:
Looks like there'll be one instance per assembly. If that's true, since it's a bounded set, it seems like we'd be better off with this as a `class` to ensure that no matter what extra fields we add to it in the future, it won't add to the cost of copying references.

I know this is `readonly` and thus often will be passed by reference, but not always.

Reply:
I kinda did this as a forcing function for it being immutable. I'm ok adding a note instead.